File: docs/src/testers/quickstart/scraping_hackernews.ts
import { z } from 'zod';
import { NotteClient } from 'notte-sdk';

const HackerNewsPost = z.object({
  title: z.string(),
  url: z.string(),
  points: z.number(),
  author: z.string(),
});

const HackerNewsFeed = z.object({
  posts: z.array(HackerNewsPost),
});

const notte = new NotteClient({
  apiKey: process.env.NOTTE_API_KEY,
});

const result = await notte.scrape('https://news.ycombinator.com', {
  instructions: 'Extract the top 5 posts from the front page',
  response_format: HackerNewsFeed,
});

result.posts.forEach((post, index) => {
  console.log(`${index + 1}. ${post.points} - ${post.title}`);
});

File: docs/src/testers/quickstart/scraping_hackernews.py
from notte_sdk import NotteClient
from pydantic import BaseModel


class HackerNewsPost(BaseModel):
    title: str
    url: str
    points: int
    author: str


class HackerNewsFeed(BaseModel):
    posts: list[HackerNewsPost]


client = NotteClient()

result = client.scrape(
    url="https://news.ycombinator.com",
    response_format=HackerNewsFeed,
    instructions="Extract the top 5 posts from the front page",
)

for i, post in enumerate(result.data.posts, 1):
    print(f"{i}. {post.points} - {post.title}")
